### ISSUE TITLE
problem with Analysis =>Wishbone

### ISSUE THREAD

**Issue by AlexSount** [ManuSetty/wishbone#68]
Hello,

I would grateful if you help me to complete the analysis of the given dataset.
Everything works fine until I try to use Analysis => Wishbone
It does not proceed and at the terminal it shows:

Using cell gate:
Gate 1
W43143
Building lNN graph...
lNN computed in : 0.02 seconds
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.6/tkinter/__init__.py", line 1705, in __call__
    return self.func(*args)
  File "/usr/local/lib/python3.6/dist-packages/wishbone-0.4.2-py3.6.egg/EGG-INFO/scripts/wishbone_gui.py", line 363, in _runWishbone
  File "/usr/local/lib/python3.6/dist-packages/wishbone-0.4.2-py3.6.egg/wishbone/wb.py", line 1068, in run_wishbone
    s=s, k=k, l=k, num_waypoints=num_waypoints, branch=branch)
  File "/usr/local/lib/python3.6/dist-packages/wishbone-0.4.2-py3.6.egg/wishbone/core.py", line 45, in wishbone
    klnn = _spdists_undirected(klnn)
  File "/usr/local/lib/python3.6/dist-packages/wishbone-0.4.2-py3.6.egg/wishbone/core.py", line 156, in _spdists_undirected
    spdists[(j,i)] = spdists[np.nonzero(spdists)]
  File "/home/alex/.local/lib/python3.6/site-packages/scipy/sparse/_index.py", line 109, in __setitem__
    raise ValueError("shape mismatch in assignment")
ValueError: shape mismatch in assignment

I use Ubuntu 18.04, python3.6 (everything seems to be updated).
I have also modified the wishbone_giu.py:
row 5: from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
and row115 (plus the other 6 identical) : self.canvas.draw()

Can you please, help because I have no idea what is wrong.

Thank you in advance!!!

Best regards,
Alex

**Comment by ManuSetty** [comment]
Can you please tell me what is the dimensions of the data you are using. Are there any columns or rows which are 0? 

Finally, can you please also try using our notebooks (Eg: https://nbviewer.jupyter.org/github/ManuSetty/wishbone/blob/master/notebooks/Wishbone_for_single_cell_RNAseq.ipynb) rather than the GUI to see if the error occurs again

**Comment by AlexSount** [comment]
Hello,

I was trying to reproduce the results of the tutorial, using the given dataset. So, there should be no problem with zeros. 

I tried to reproduce the notebook for sc-RNA Seq data, using the jupyter qtconsole. It worked when I started from the clean Ubuntu installation (otherwise I faced the previously reported error "wb" error at [2]). 
I have an issue regarding the save location of the GSEA results, but I will try to figure out where are they.

Thank you very much for the useful response.  

All the best,
Alex
